src/Repository/RepositoryMapUpdater.php

```diff
@@ -0,0 +1,150 @@
+        $this->repositoryReader = $repositoryReader;
+        $this->packagistReader = $packagistReader;
+    }
+
+    /**
+     * Reads the repositories name of each repository in the list
+     *
+     * @param string $userName   Owner of the repositories
+     * @param bool   $reloadList Do we need to reload list of repositories?
+     * @param int    $limit      Number of repositories names to read
+     */
+    public function update(string $userName, bool $reloadList, int $limit)
+    {
+        // Number of updated repositories.
+        $updated = 0;
+        // Number of skipped repositories.
+        $skipped = 0;
+
+        // Reload repositoryList to get repositories not mapped yet.
+        if ($reloadList) {
+            $this->logger->info($this->reloadList($userName) . ' repositories added.');
+        }
+
+        // Reads either local or distant repository list.
+        $repositoryList = $this->repositoryMapAccessor->read();
+
+        // Displays the number of repositories to update.
+        $toUpdate = 0;
+        foreach ($repositoryList as $repository) {
+            /** @var Repository $repository */
+            if ($repository->getExtensionName() === '') {
+                $toUpdate++;
+            }
+        }
+        $this->logger->info(
+            $toUpdate . ' repositories(s) to analyze' . ($limit ? ' (limited to ' . $limit . ')' : '') . '.'
+        );
+
+        // Finds all repository that have an empty extension name.
+        foreach ($repositoryList as $repositoryName => &$repository) {
+            /** @var Repository $repository */
+            if ($repository->getExtensionName() === '') {
+                $this->logger->info('Analyzing repository "' . $repositoryName . '"...');
+                $repository = $this->repositoryReader->analyzeRepository($repository);
+
+                if (--$limit === 0) {
+                    break;
+                }
+            }
+        }
+
+        // Persists updated repositoryList.
+        $this->repositoryMapAccessor->write($repositoryList);
+
+        // Displays results.
+        if ($updated) {
+            $this->logger->info($updated . ' repositories updated.');
+        }
+        if ($skipped) {
+            $this->logger->info($skipped . ' repositories skipped.');
+        }
+    }
+
+    /**
+     * Reads list of repositories from GitHub.
+     *
+     * @param string $owner
+     *
+     * @return int Number of repositories added.
+     */
+    public function reloadList(string $owner)
+    {
+        $organizationProperties = $this->repositoryReader->getOrganizationProperties($owner);
+        $message = 'Connected to GitHub with http token.' . "\n"
+            . 'Organisation "' . $owner . '" has:' . "\n"
+            . '- ' . $organizationProperties['public_repos'] . ' public repositories' . "\n"
+            . '- ' . $organizationProperties['total_private_repos'] . ' private repositories' . "\n";
+        echo $message;
```

Comment:
You should not write anything to output on the service level, please bubble up the message and display by the command